src/helpers/MappingHelpers.ts

```diff
@@ -0,0 +1,86 @@
+import {
+    DependencyLink,
+    DependencyNode,
+    Service,
+    ServiceDto,
+} from '../components/types';
+
+export function mapServiceDtoToService(serviceDto: ServiceDto): Service {
+    const service: Service = {
+        name: serviceDto.name,
+        version: serviceDto.version,
+        capabilities: {},
+        expectations: {},
+    };
+
+    const { expectations, capabilities } = serviceDto;
+
+    const providerServices = Object.keys(expectations);
+
+    providerServices.forEach((serviceName: string) => {
+        service.expectations[serviceName] = {};
+        const types = Object.keys(expectations[serviceName]);
+
+        types.forEach((type: string) => {
+            service.expectations[serviceName][type] = {
+                ...expectations[serviceName][type],
+                value: JSON.parse(expectations[serviceName][type].value),
+            };
+        });
+    });
+
+    const capabilityTypes = Object.keys(capabilities);
+
+    capabilityTypes.forEach((type: string) => {
+        service.capabilities[type] = {
+            ...capabilities[type],
+            value: JSON.parse(capabilities[type].value),
+        };
+    });
+
+    return service;
+}
+
+export function createLinks(
+    nodes: DependencyNode[],
+    services: Service[]
+): DependencyLink[] {
+    return services.reduce((links: DependencyLink[], service: Service) => {
+        const linksWithTypes: DependencyLink[] = [];
+        const providerNames = Object.keys(service.expectations);
+
+        providerNames.forEach((name: string) => {
+            const sourceNode = nodes.find(
+                (node: DependencyNode) =>
+                    node.name === service.name &&
+                    node.version === service.version
+            ) || {
+                name: '',
```

Comment:
we should not continue when we have empty name or empty version
because we should not a lot of additional computations, when we know that this ``sourceCode`` can't be link